### DISCUSSION TITLE
zusätzliche Werte

### DISCUSSION THREAD

**Discussion by petrini01** [weltenwort/home-assistant-rct-power-integration#234] [Ideas]
Erst einmal vielen Dank für die Integration des RCT Power für Home Assitant. Es hat bei mir sofort geklappt und liefert gute Werte. Seitens RCT habe ich nach Rückfrage das Datenprotokoll V1.14 erhalten. Dabei vielen mir einige Werte auf, welche bisher noch nicht in der Datenschnittstelle enthalten sind. Dabei geht es mir für die weitere Haussteuerung (Freigabe einer Wärempumpe) um die Erkennung von Inselbetrieb und Batteriekalibrierung. Ist es möglich, weitere Werte in die Datenschnittstelle zu integrieren? Für die schnelle Erkennung von Insel- oder Netzbetrieb wäre es ebenfalls wünschenswert, ein kleineres Pollintervall (1 bis 2 Sekunden) für  bestimmte Statusinformationen zu ermöglichen.

0x3623D82A, // 182: prim_sm.island_flag | t_uint16 | 
Grid-separated

0x7AF779C1, // 443: nsm.pu_mode | t_bool | 
P(U) mode 0: Pn 1: Pload

0x81AF854E, // 466: nsm.pu_use | t_bool | 
P(U) active

0x97E203F9, // 554: power_mng.is_grid | t_bool | 
power_mng.is_grid

0xC40D5688, // 694: prim_sm.state_source | t_uint32 | 
prim_sm.state_source

0xC9900716, // 717: power_mng.is_island_only | t_bool | 
Island without power switch support

0xDC667958, // 770: power_mng.state | t_uint8 | 
Battery state machine

0xFB5DE9C5, // 873: prim_sm.minigrid_flag | t_bool | 
Minigrid support

0xFD72CC0D, // 887: frt.enabled | t_bool | 
Enable FRT

Es wäre gut, wenn der Status rct_fault ebenfalls in einem kleineren Intervall abgefragt werden könnte.

Für die detailiertere Überwachung der Gesamtanlage sind mir noch folgende Werte aufgefallen. Können diese mit integriert werden?

0x8EBF9574, // 506: power_mng.soc_min_island | t_float | 
Min SOC target (island)

0xB6623608, // 644: power_mng.bat_next_calib_date | t_uint32 | 
Next battery calibration

0xC4D87E96, // 697: prim_sm.island_retrials | t_uint16 | 
Island trials counter

0xF393B7B0, // 846: power_mng.calib_charge_power | t_float | 
Calibration charge power [W]

Für manche der o.g. Statusinformationen gibt es noch keine offizielle Beschreibung. Zusammen mit einem kleinen Pollintervall läßt sich dieses jedoch unter Betrachtung des aktuellen Anlagenzustandes bestimmt ergänzen und den anderen Usern zur Verfügung stellen.

Danke und viele Grüße
Peter

**Comment by weltenwort** [comment]
Hi Peter,

danke für die ausführliche Beschreibung. Die neuen Datenpunkte hinzuzufügen dürfte kein Problem sein. Das kleine Abfrageinterval könnte allerdings zu mehr Fehlern in dem nicht gerade robusten Netzwerkprotokoll führen. :thinking: Vielleicht müssten wir dafür das scheduling etwas aufbohren. Ich werde basierend auf diesem Thread mal ein paar Issues anlegen.

---

Hi Peter,

thank you for the detailed description. Adding the new data points shouldn't be a problem. The even-smaller polling interval, though, might increase the risk of collisions and read errors. :thinking: Maybe we need a more elaborate scheduling approach to mitigate that. I'll create a few issues based on your requests.
